compositor_render/src/render_loop.rs

```diff
@@ -2,36 +2,52 @@ use std::{collections::HashMap, sync::Arc, time::Duration};
 
 use compositor_common::{
     scene::{InputId, OutputId},
+    util::RGBColor,
     Frame,
 };
 use log::error;
 
-use crate::renderer::{
-    scene::{Node, Scene},
-    RenderCtx,
+use crate::{
+    frame_set::FrameSet,
+    renderer::{
+        scene::{Node, Scene},
+        RenderCtx,
+    },
 };
 
 pub(super) fn populate_inputs(
     ctx: &RenderCtx,
     scene: &mut Scene,
-    frames: &mut HashMap<InputId, Frame>,
+    frame_set: &mut FrameSet<InputId>,
 ) {
-    for (input_id, (node, input_textures)) in &mut scene.inputs {
-        let Some(frame) = frames.remove(input_id) else {
+    for (input_id, (_, input_textures)) in &mut scene.inputs {
+        let Some(frame) = frame_set.frames.remove(input_id) else {
+            input_textures.clear();
             continue;
         };
-        node.output.ensure_size(ctx.wgpu_ctx, frame.resolution);
+        if Duration::saturating_sub(frame_set.pts, ctx.stream_fallback_timeout) > frame.pts {
+            input_textures.clear();
+            continue;
+        }
+
```

Comment:
I don't like the idea of clearing input textures without logging anything / notifying the user. It makes it harder to find whether you are getting the black screen on the output due to some rendering or network issue.

Reply:
What would you suggest?
- We can't log it here every time frame is missing because it produce to much logs
- We don't really know what is the problem here regardless. We can only distinguish missing frames and frames that are too old.

Reply:
We could keep track of which inputs we cleared last time and log only new ones. It would reduce log count, while we could still notify the user, that some frames are missing / too old.

Reply:
add issue https://github.com/membraneframework/video_compositor/issues/102